### ISSUE TITLE
Vroom 1.6.0 fails when passed list of (curl) connections

### ISSUE THREAD

**Issue by Syntopia** [tidyverse/vroom#469]
After upgrading to Vroom 1.6.0, it seems it is no longer possible to pass a list of curl connections - this used to work on 1.5.7. 

Passing a list of URLs still works.

```
library(curl)
library(vroom)
url = "https://people.sc.fsu.edu/~jburkardt/data/csv/addresses.csv"

# Fails - worked with Vroom 1.5.7
t <- vroom(list(curl(url),curl(url)))

# Works
t <- vroom(c(url,url))
```

The error message in 1.6.0 is:
```
Error in enc2utf8(path) : argument is not a character vector
```

**Comment by hadley** [comment]
I think this must have only worked by accident; it's certainly not documented.

**Comment by jennybc** [comment]
From my reading, vroom does actually claim to read a list of connections, in both its README and in its Get Started vignette. I'm not re-opening this, though, because it's basically a duplicate of #509, which will soon be addressed by #514.
